desktop/src/gui/bookmarks_dialog.rs

```diff
@@ -0,0 +1,171 @@
+                                        .wrap(false),
+                                );
+                            });
+                            row.col(|ui| {
+                                ui.add(
+                                    Label::new(bookmark.url.as_str())
+                                        .selectable(false)
+                                        .wrap(false),
+                                );
+                            });
+
+                            let response = row.response();
+                            response.context_menu(|ui| {
+                                if ui.button(text(locale, "remove")).clicked() {
+                                    ui.close_menu();
+                                    action = Some(BookmarkAction::Remove(index));
+                                }
+                            });
+                            if response.clicked() {
+                                self.selected_bookmark = Some(SelectedBookmark {
+                                    index,
+                                    // TODO: set hint
+                                    url: PathOrUrlField::new(Some(bookmark.url.clone()), ""),
+                                });
+                            }
+                        });
+                    }
+                });
+            });
+
+        if let Some(action) = action {
+            if let Err(e) = self.preferences.write_bookmarks(|writer| match action {
+                BookmarkAction::Remove(index) => {
+                    self.selected_bookmark = None;
+                    writer.remove(index);
+                }
+            }) {
+                tracing::warn!("Couldn't update bookmarks: {e}");
+            }
+        }
+    }
+
+    fn show_bookmark_panel(&mut self, locale: &LanguageIdentifier, ui: &mut Ui) {
+        if let Some(bookmark) = &mut self.selected_bookmark {
+            Grid::new("bookmarks-dialog-panel-grid")
+                .num_columns(2)
+                .show(ui, |ui| {
+                    let previous_url = bookmark.url.value().cloned();
+
+                    ui.label(text(locale, "bookmarks-dialog-url"));
+                    let current_url = bookmark.url.ui(locale, ui).value();
+
+                    if previous_url.as_ref() != current_url {
+                        if let Some(url) = current_url {
+                            if let Err(e) = self.preferences.write_bookmarks(|writer| {
+                                writer.set_url(bookmark.index, url.clone());
+                            }) {
+                                tracing::warn!("Couldn't update bookmarks: {e}");
+                            }
+                        }
+                    }
+
+                    ui.end_row();
+                });
+        } else {
+            ui.vertical_centered_justified(|ui| {
+                ui.label(text(locale, "bookmarks-dialog-not-selected"));
```

Comment:
Ohh is this the "Nothing to show" in your screenshot? :sweat_smile:

Can we just have a button to edit (next to each bookmark) and reuse the dialog I proposed in an earlier review comment?

Reply:
If we just add an initial add dialog, not sure if need to change anything here, maybe just the translation to be more concrete (e.g. `No bookmark selected`)